react/src/components/QrReaderOverlay/QrReaderOverlayContainer.tsx

```diff
@@ -251,18 +207,65 @@ function QrReaderOverlayContainer({
     [qrValueResolver],
   );
 
+  const handleSingleScan = useCallback(
+    (result: string) => {
+      const qrCode = extractQrCodeFromUrl(result);
+      if (qrCode == null) {
+        onScanningDone([{ kind: QrResolverResultKind.NOT_BOXTRIBUTE_QR }]);
+      } else {
+        apolloClient
+          .query<GetBoxLabelIdentifierForQrCodeQuery, GetBoxLabelIdentifierForQrCodeQueryVariables>(
+            {
+              query: GET_BOX_LABEL_IDENTIFIER_BY_QR_CODE,
+              fetchPolicy: "no-cache",
+              variables: { qrCode },
+            },
+          )
+          .then(({ data, errors }) => {
+            if ((errors?.length || 0) > 0) {
+              const errorCode = errors ? errors[0].extensions.code : null;
+              if (errorCode === "FORBIDDEN") {
+                onScanningDone([
+                  { kind: QrResolverResultKind.NOT_AUTHORIZED, qrCodeValue: qrCode },
+                ]);
+              } else if (errorCode === "BAD_USER_INPUT") {
+                onScanningDone([
+                  { kind: QrResolverResultKind.NOT_BOXTRIBUTE_QR, qrCodeValue: qrCode },
+                ]);
+              } else {
+                onScanningDone([{ kind: QrResolverResultKind.FAIL, qrCodeValue: qrCode }]);
```

Comment:
You could add another test case with error code `INTERNAL_SERVER_ERROR` to cover this line and the one below.

Reply:
done